ex2

evkingen opened the pull request.

evkingen committed:
NewsActivity - добавлены кнопка Send Message, иконки обратной связи.
activity_news.xml - разметка для потретной ориентации.
layout-land\activity_news.xml - разметка для альбомной ориентации.
detail_description.xml - используется в разметках activity_news. Отображает детальную информацию.
Обновлены файлы с ресурсами.

evkingen committed:
Доработка по общим замечаниям.
В разметках изменены отступы от границ экрана с margin на padding
Переписан detal_description.xml с RelativeLayout на LinearLayout
Уменьшены иконки
Доработана кнопка SEND MESSAGE

evkingen committed:
небольшие правки по коду

evkingen committed:
небольшая правка по коду

colriot commented on app/src/main/java/com/alohagoha/newsapp/NewsActivity.java:
```diff
@@ -1,13 +1,57 @@
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_news);
+        final EditText et = findViewById(R.id.feedback_et);
+        Button btn_send = findViewById(R.id.send_btn);
```
Названия переменных в Java принято задавать camelCase, не через подчёркивание

evkingen replied:
Переименовал в camelCase на более осмысленные

colriot commented on app/src/main/java/com/alohagoha/newsapp/NewsActivity.java:
```diff
@@ -1,13 +1,57 @@
+        intent.putExtra(Intent.EXTRA_EMAIL, new String[] {EMAIL_ADDRESS});
+        intent.putExtra(Intent.EXTRA_SUBJECT,EMAIL_SUBJECT);
+        intent.putExtra(Intent.EXTRA_TEXT,"Text1 " + emailText);
+        startActivity(intent);
```
Работает ли такой способ отправки email-ов в реальности на твоём девайсе? Все ли поля проставлены, только ли почтовые клиенты предлагаются для обработки?

evkingen replied:
Способ работал, поля проставлялись, но предлагалось открыть куча приложений, Подглядел - сделал через ACTION_SENDTO

colriot replied:
Да, к сожалению, чтобы именно емейл отправить, приходится работать с верным url

colriot commented on app/src/main/res/layout-land/activity_news.xml:
```diff
@@ -0,0 +1,29 @@
+        android:layout_width="0dp"
+        android:layout_height="match_parent"
+        android:layout_weight="1"
+        android:src="@drawable/iii"
```
Что за названия? Непонятно, что это за картинки. Почему их три? И почему они такое громадные? 29 + 12 + 2 МБ. Для почти пустого приложения - это гигантские размеры. Юзеры обычно не хотят качать такое, когда неочевидна причина. Поэтому мы и боремся за минимальный размер.

evkingen replied:
Удалил лишние ресурсы, названия переименовал.

colriot commented on app/src/main/res/layout-land/activity_news.xml:
```diff
@@ -0,0 +1,29 @@
+<?xml version="1.0" encoding="utf-8"?>
```
А почему `activity_news`? Вроде с новостями не связано

evkingen replied:
Переименовал) activity_news будет следующим.

colriot commented on app/src/main/res/layout-land/activity_news.xml:
```diff
@@ -0,0 +1,29 @@
+        android:paddingEnd="@dimen/spacing_default"
+        android:layout_width="0dp"
+        android:layout_height="wrap_content"
+        android:layout_weight="2">
```
Не забывай форматировать код перед заливкой. А лучше всегда. Форматнёшь и атрибуты станут в правильной последовательности.

evkingen replied:
Вроде как отформатировал все .xml с помощью Reformat

colriot commented on app/src/main/res/layout-land/activity_news.xml:
```diff
@@ -0,0 +1,29 @@
+        <include layout="@layout/detail_description"/>
+    </ScrollView>
+
+</LinearLayout>
```
Лучше завести себе привычку всегда иметь пустую строку в конце файла, чтобы не было таких красных маркеров, и при изменениях в последних строках в диффа (git-а) не было мусора.
Касается всех файлов проекта.

evkingen replied:
надобавлял пустых строк. завел привычку :)

colriot commented on app/src/main/res/layout/activity_news.xml:
```diff
@@ -1,18 +1,28 @@
+        android:paddingTop="@dimen/spacing_small"
+        android:paddingStart="@dimen/spacing_start"
+        android:paddingEnd="@dimen/spacing_default"
         android:layout_width="wrap_content"
```
По форматированию коммент тоже касается всего проекта

evkingen replied:
отформатировал

colriot commented on app/src/main/res/layout/detail_description.xml:
```diff
@@ -0,0 +1,105 @@
+        android:textSize="@dimen/size_text"
+        android:textColor="@android:color/holo_red_light"
+        android:text="@string/button_send_message"
+        android:layout_gravity="right"
```
`right` -> `end`   Мы хотим поддерживать right-to-left локали)

evkingen replied:
переделал

colriot commented on app/src/main/res/layout/detail_description.xml:
```diff
@@ -0,0 +1,105 @@
+        android:textColor="@android:color/holo_red_light"
+        android:text="@string/button_send_message"
+        android:layout_gravity="right"
+        style="@style/Widget.AppCompat.Button.Borderless"/>
```
👍

colriot commented on app/src/main/res/layout/detail_description.xml:
```diff
@@ -0,0 +1,105 @@
+        android:layout_height="1dp"
+        android:layout_marginTop="@dimen/spacing_small"
+        android:layout_marginBottom="@dimen/spacing_small"
+        android:background="?android:attr/listDivider" />
```
👍

colriot commented on app/src/main/res/layout/detail_description.xml:
```diff
@@ -0,0 +1,105 @@
+        android:layout_marginBottom="@dimen/spacing_small"
+        android:text="@string/main_information"
+        android:textSize="@dimen/size_text" />
+    <View
```
Лучше вынести кусок с разделителем в отдельный файл и инклюдить там, где понадобится

evkingen replied:
вынес в divider.xml

colriot commented on app/src/main/res/layout/detail_description.xml:
```diff
@@ -0,0 +1,105 @@
+            android:layout_width="@dimen/size_icon"
+            android:layout_height="@dimen/size_icon"
+            android:layout_margin="@dimen/spacing_small"
+            android:src="@drawable/instag"
```
Не обрубай названия без надобности (если это не общепринятые аббревиатуры, или хотя бы принятые на уровне твоей команды), экономия 3 букв ничего не даст, кроме потенциальной путаницы

evkingen replied:
переименовал

colriot commented on build.gradle:
```diff
@@ -5,10 +5,10 @@ buildscript {
     repositories {
         google()
         jcenter()
+        mavenCentral()
```
А зачем это добавил?

evkingen replied:
Я добавлял ретролямбду кажется. Потом понял что она не нужна.

colriot commented on app/src/main/res/values/dimens.xml:
```diff
@@ -0,0 +1,8 @@
+    <dimen name="size_text">16sp</dimen>
+    <dimen name="size_icon">28dp</dimen>
+    <dimen name="spacing_default">16dp</dimen>
+    <dimen name="spacing_start">50dp</dimen>
```
В следующих домашках это хорошо было бы переименовать, потому что это актуально только для этого экрана.

colriot commented on app/src/main/res/values/strings.xml:
```diff
@@ -1,4 +1,17 @@
+    <string name="desc_java">desc_java</string>
+    <string name="learn_java">Learn Java</string>
+    <string name="main_information">This information about Georgiy. Soon.</string>
+    <string name="hint_et">Enter a message...</string>
```
Среда должна выделять это жёлтым, предлагать заменить три точки на троеточие. Если нажать `Alt+Enter`, то она предложит автоисправление.

colriot replied:
ET - непонятно, не для любого EditText же? Лучше конкретно указать, что это хинт почтового сообщения

colriot commented on app/src/main/res/values/strings.xml:
```diff
@@ -1,4 +1,17 @@
     <string name="hello_world">Hello World</string>
+    <string name="key_info_1">IBM Notes Developer in Airport Domodedovo</string>
+    <string name="my_photo">my photo</string>
+    <string name="desc_notes9">desc_notes9</string>
```
Что значит 9?

colriot replied:
И значение вообще ничего не объясняет

evkingen replied:
Переименовал переменные и значения на более осмысленные.

colriot commented on app/src/main/res/values/strings.xml:
```diff
@@ -1,4 +1,17 @@
+    <string name="key_info_1">IBM Notes Developer in Airport Domodedovo</string>
+    <string name="my_photo">my photo</string>
+    <string name="desc_notes9">desc_notes9</string>
+    <string name="desc_magic">desc_magic</string>
```
Лучше чуть более понятно написать в ключе. Со значением то же самое

evkingen replied:
переименовал название переменных и значение.

colriot commented on app/src/main/res/values/strings.xml:
```diff
@@ -1,4 +1,17 @@
+    <string name="desc_notes9">desc_notes9</string>
+    <string name="desc_magic">desc_magic</string>
+    <string name="key_info_2">Magic: The Gathering player</string>
+    <string name="desc_java">desc_java</string>
```
Опять

evkingen replied:
:) переименовал.

colriot commented on app/src/main/res/values/strings.xml:
```diff
@@ -1,4 +1,17 @@
+    <string name="main_information">This information about Georgiy. Soon.</string>
+    <string name="hint_et">Enter a message...</string>
+    <string name="button_send_message">SEND MESSAGE</string>
+    <string name="descr_instagram_ref">instagram ref</string>
```
Та же история с этими описаниями

evkingen replied:
Описания переименованы.

colriot commented on app/src/main/res/values/styles.xml:
```diff
@@ -8,4 +8,7 @@
         <item name="colorAccent">@color/colorAccent</item>
     </style>
 
+    <style name="MyButton" parent="Theme.AppCompat.Light">
```
Стиль нигде не используется, можно удалить.

evkingen replied:
Удалил

colriot commented on app/src/main/res/layout/detail_description.xml:
```diff
@@ -0,0 +1,105 @@
+        android:gravity="center"
+        android:orientation="horizontal">
+        <ImageView
+            android:id="@+id/iv_ref_1"
```
Лучше более понятные названия дать. По сути, это кнопки для соц аккаунтов, так и можно их назвать, с использованием названий сетей.

evkingen replied:
все id переименовал

colriot reviewed (commented):
Есть пачка замечаний :)

colriot commented on app/build.gradle:
```diff
@@ -16,12 +16,17 @@ android {
 }
 
 dependencies {
     implementation fileTree(dir: 'libs', include: ['*.jar'])
```
Реально у тебя нет такой папки и библиотек в ней. Стоит удалить это, чтобы было меньше мусора

evkingen replied:
убрал

colriot commented on app/build.gradle:
```diff
@@ -16,12 +16,17 @@ android {
     implementation 'com.android.support:appcompat-v7:28.0.0'
     implementation 'com.android.support.constraint:constraint-layout:1.1.3'
+
     testImplementation 'junit:junit:4.12'
```
Так как реальных тестов нет, то лучше удалить и пустые классы с пустыми тестами, и зависимости для тестов.

evkingen replied:
убрал

evkingen committed:
Удалены большие лишние изображения.

evkingen committed:
Коррекция размера изображений.

evkingen committed:
Замечание по грэдлу. Убраны лишние зависимости.

evkingen committed:
Переименовано название активити. Переименованы переменные. Изменена функция sendMail: intent предлагает на открытие только почтовое предложение.

evkingen committed:
Правки по макету:
форматирование кода,
переименование переменных,
изменены значения описаний imageview,
divider вынесен в отдельный файл.

evkingen committed:
email_subject перенесен в строковый ресурс

evkingen committed:
добавлен модификатор final к read-only переменным

evkingen commented:
![screenshot_2018-11-01-13-35-10-915_com alohagoha newsapp 1](https://user-images.githubusercontent.com/38916658/47847870-9e691d80-dddd-11e8-8c24-461465bd8c30.png)
![screenshot_2018-11-01-13-35-19-998_com alohagoha newsapp 1](https://user-images.githubusercontent.com/38916658/47847886-aaed7600-dddd-11e8-9556-d5f86962fbec.png)
![screenshot_2018-11-01-13-35-27-894_com alohagoha newsapp 1](https://user-images.githubusercontent.com/38916658/47847892-afb22a00-dddd-11e8-8315-b593ffcfba0e.png)
![screenshot_2018-11-01-13-36-43-998_com alohagoha newsapp 1](https://user-images.githubusercontent.com/38916658/47847901-b80a6500-dddd-11e8-9436-b56df43e1b5b.png)
![screenshot_2018-11-01-13-36-50-640_com alohagoha newsapp 1](https://user-images.githubusercontent.com/38916658/47847906-bccf1900-dddd-11e8-8783-6addd17cc09a.png)
![screenshot_2018-11-01-13-36-53-412_com alohagoha newsapp 1](https://user-images.githubusercontent.com/38916658/47847914-c35d9080-dddd-11e8-8b11-044254a3d7b6.png)

colriot reviewed (approved):
Отлично! Поехали дальше :)
